lib/krikri/harvester.rb

```diff
@@ -100,9 +100,9 @@ def get_record(_)
     # This should be idempotent so it can be safely retried on errors.
     #
     # @return [Boolean]
-    def run
+    def run(activity_uri = nil)
       log :info, 'harvest is running'
-      records.each(&:save)
+      records.each { |rec| rec.save(activity_uri) }
```

Comment:
Not through `#run` (and therefore not through the `Job` queue). You could do it various other ways, and still save with an `activity_uri`.

As you seem to be pointing out, this change doesn't impact that issue either way.